### ISSUE TITLE
Opening image and close it redirects to files app

### ISSUE THREAD

**Issue by MorrisJobke** [nextcloud/files_filter#10]
* upload some images and enable gallery app
* open the images filter
* click on one image -> gallery slideshow opens
* click on close
* expected: being back in the images filter
* actual: being back in the normal "all files" view

This could also be caused by the refresh of the gallery app on close. But I couldn't find the issue anymore. 

cc @icewind1991 @rullzer @oparoz

**Comment by oparoz** [comment]
AFAIK, Gallery, just uses History.back()

**Comment by MorrisJobke** [comment]
> AFAIK, Gallery, just uses History.back()

That would not allow to close the view. It would basically go backwards in the slideshow, right?

**Comment by oparoz** [comment]
No because the slideshow is using # navigation, which is not added to History.

**Comment by MorrisJobke** [comment]
> No because the slideshow is using # navigation, which is not added to History.

You are right: https://github.com/nextcloud/gallery/blob/3d542dba41379bce91f153fd31f3e7b989db9e0d/js/slideshowcontrols.js#L405-L405

**Comment by MorrisJobke** [comment]
> You are right: https://github.com/nextcloud/gallery/blob/3d542dba41379bce91f153fd31f3e7b989db9e0d/js/slideshowcontrols.js#L405-L405

Changing this to the `stop()` call fixes the problem for me. What is the reason for the `History.back()`, @oparoz ?

**Comment by oparoz** [comment]
Because there are 3 ways to stop the slideshow. Back, esc and pressing on X. They should all produce the same result.

**Comment by MorrisJobke** [comment]
> Because there are 3 ways to stop the slideshow. Back, esc and pressing on X. They should all produce the same result.

And calling `this.stop()` is fine? https://github.com/nextcloud/gallery/blob/3d542dba41379bce91f153fd31f3e7b989db9e0d/js/slideshowcontrols.js#L409

**Comment by oparoz** [comment]
It could, yes, but I'm guessing it offers a degraded experience, like loss of history. So not sure if it's the best option...

**Comment by MorrisJobke** [comment]
> It could, yes, but I'm guessing it offers a degraded experience, like loss of history. So not sure if it's the best option...

But the history didn't work anyway that good, or am I wrong?

**Comment by MorrisJobke** [comment]
@nextcloud/javascript Other ideas?

**Comment by oparoz** [comment]
I think this is the real problem:
https://github.com/nextcloud/gallery/blob/3d542dba41379bce91f153fd31f3e7b989db9e0d/js/galleryfileaction.js#L178-L181

**Comment by nickvergessen** [comment]
Dropping https://github.com/nextcloud/gallery/blob/3d542dba41379bce91f153fd31f3e7b989db9e0d/js/slideshowcontrols.js#L406 solves this issue.

However since the favorite filter is also affected, this is either caused by the framework of the files app or by the gallery.
At least not by this app.

**Comment by nickvergessen** [comment]
Also the gallery app should use our OC.Util.History for handling instead of doing its stuff directly.

**Comment by MorrisJobke** [comment]
> Dropping https://github.com/nextcloud/gallery/blob/3d542dba41379bce91f153fd31f3e7b989db9e0d/js/slideshowcontrols.js#L406 solves this issue.
> 
> However since the favorite filter is also affected, this is either caused by the framework of the files app or by the gallery.
> At least not by this app.

This means that the files filter does not push the latest URL to the history. The URL for the slideshow is missing. Should this be done in gallery or in the files filter?
